spring-kafka/src/test/java/org/springframework/kafka/listener/ConcurrentMessageListenerContainerTests.java

```diff
@@ -142,7 +145,7 @@ protected Consumer<Integer, String> createKafkaConsumer(String groupId, String c
 
 		final CountDownLatch latch = new CountDownLatch(3);
 		final Set<String> listenerThreadNames = new ConcurrentSkipListSet<>();
-		final List<String> payloads = new ArrayList<>();
+		final List<String> payloads = Collections.synchronizedList(new ArrayList<>());
```

Comment:
Any particular reason to wrap?

Reply:
We need a synchronized ArrayList. This is shared in all the MessageListener instances.

Reply:
And why is that?
What is the problem with non-synchornized here?
We just add data into the list and then we verify it like this:
```
assertThat(payloads).containsExactlyInAnyOrder("foo", "bar", "qux");
```

So, no matter who and when adds items to the list, we are just save according to our expectations.

Reply:
Might be, but the code is like that for years and we haven't seen any errors yet.
If you say it is possible, I would leave it as is and prefer to see that error eventually.
How this change is relevant to the problem of the PR?
Why make changes which are going to be hard to track in the future?
The contributors come and go, but the code is on us to support, therefore we want to be confident what we are going to deal with.

Reply:
It is possible for the error since ArrayList add method is not thread-safe. Please let me know if you want me to change that to plain ArrayList. I will do that.

Reply:
That's correct, but we are not in that situation in this logic. See Javadoc of `ArrayList`:
```
 * <p><strong>Note that this implementation is not synchronized.</strong>
 * If multiple threads access an {@code ArrayList} instance concurrently,
 * and at least one of the threads modifies the list structurally, it
 * <i>must</i> be synchronized externally.  (A structural modification is
 * any operation that adds or deletes one or more elements, or explicitly
 * resizes the backing array; merely setting the value of an element is not
 * a structural modification.)  This is typically accomplished by
 * synchronizing on some object that naturally encapsulates the list.
```
>  merely setting the value of an element is not a structural modification

So, just adding as we do here is OK.

Please, revert this change in favor of the concurrent performance and possible error bubbling in the future for our education.